### ISSUE TITLE
Users should only be able to register for an event once (by default)

### ISSUE THREAD

**Issue by BenJam** [opencollective/opencollective#5086]
## User story 
As a event organiser I want attendees to only register once for an event

## Notes
It would appear that this _was_ a feature at one point, but it is no longer available in this drop-down on the create event form:

![Screenshot 2022-01-04 at 09 36 17](https://user-images.githubusercontent.com/158833/148041225-45e4874f-7f18-4c98-be2a-634a6ab9d0c1.png)

this leads to the very confusing situation in which an event can have 8 of 8 tickets 'sold' but only seven attendees... I have had to adjust the number of tickets in order to allow an 8th person to register at https://opencollective.com/workshops/events/recruiting-financial-sponsors-b4db9a41 for example

**Comment by SudharakaP** [comment]
I was looking at this and I was wondering why we have multiple ticket ordering in the first place. When the user orders a ticket they get to choose the quantity;

![image](https://user-images.githubusercontent.com/12435965/148503414-f7b36b27-1708-44bb-8af2-66233ee948b8.png)

However in the contribution flow they don't specify who the tickets are for. And current flow seem to only allow same user to book multiple tickets which maybe doesn't make sense. If we allow multiple tickets maybe we need a mechanism to specify to whom the tickets are booked for?  

Alternatively as a simple fix for this issue maybe I suggest removing this multiple ticket booking? So that users can only book 1 ticket per user. If they need to book multiple tickets they need to have multiple accounts? WDYT? Maybe I am missing something here. 🤔 

cc: @znarf @Betree

**Comment by Betree** [comment]
Asking for a name for each ticket could be interesting, but it's not the root cause for this issue. When you book tickets for an event (concert, theater, etc) the website often asks for a single name that serves for the reservation, even if you book multiple seats. I just checked on [Ticketmaster](https://www.ticketmaster.com/), and they don't seem to be asking for a second name when you book 2 tickets.

So TLDR ; it's fine and pretty standard to allow users to book multiple tickets in one go, we should not break that feature. We can eventually implement a new tier setting to only allow booking one ticket per user, but it should not impact existing tiers.

**Comment by SudharakaP** [comment]
> So TLDR ; it's fine and pretty standard to allow users to book multiple tickets in one go, we should not break that feature. We can eventually implement a new tier setting to only allow booking one ticket per user, but it should not impact existing tiers.

Thanks for the ideas. So as for the implementation here's a suggestion, we can store if it's single or multiple tickets within the `Tier.data` column ({ singleTicket: true }) and then make sure we don't show the quantity selector when booking if it's a single ticket. Does that sound good?

**Comment by Betree** [comment]
> > So TLDR ; it's fine and pretty standard to allow users to book multiple tickets in one go, we should not break that feature. We can eventually implement a new tier setting to only allow booking one ticket per user, but it should not impact existing tiers.
> 
> Thanks for the ideas. So as for the implementation here's a suggestion, we can store if it's single or multiple tickets within the `Tier.data` column ({ singleTicket: true }) and then make sure we don't show the quantity selector when booking if it's a single ticket. Does that sound good?

No blocker for me. It's really MVP, as users will still be able to go through the flow again.

@BenJam To better understand the use case described in https://github.com/opencollective/opencollective/issues/5086#issue-1093186021, do you know why this person booked 8 tickets in the first place? It sounds like a valid use case to be able to book additional tickets for the people that will come with you.

**Comment by SudharakaP** [comment]
> > > So TLDR ; it's fine and pretty standard to allow users to book multiple tickets in one go, we should not break that feature. We can eventually implement a new tier setting to only allow booking one ticket per user, but it should not impact existing tiers.
> > 
> > 
> > Thanks for the ideas. So as for the implementation here's a suggestion, we can store if it's single or multiple tickets within the `Tier.data` column ({ singleTicket: true }) and then make sure we don't show the quantity selector when booking if it's a single ticket. Does that sound good?
> 
> No blocker for me. It's really MVP, as users will still be able to go through the flow again.

If they use the same login and if it's a single ticket we can hide the ticket for the event as well ? Of course if they really want they can use a different login or book another ticket as a guest.

**Comment by Betree** [comment]
> If they use the same login and if it's a single ticket we can hide the ticket for the event as well ? Of course if they really want they can use a different login or book another ticket as a guest.

We can already see the complexity quickly growing from there. I'd like to make sure we clearly understand the problem and propose a future-proof solution before we jump on implementation. Let's wait for @BenJam's feedback.

**Comment by SudharakaP** [comment]
This was also requested in the Fresh-desk ticket; https://opencollective.freshdesk.com/a/tickets/125729. 

Looking at the ticket and the former comments on this thread, I suggest the following. 

1) Store if it's single or multiple tickets within the `Tier.data` column (`{ singleTicket: true }`) and then make sure we don't show the quantity selector when booking if it's a single ticket.

2) We also add a checkbox to the Tickets settings; `Allow only registered users`. If this is selected the ticket can only be purchased by registered users. This way the organizer of the event can prevent people from booking tickets as guests thus making sure they know who is coming exactly to the event. 

@BenJam : Let us know what you think. 

cc: @Betree

**Comment by Betree** [comment]
@SudharakaP I don't understand how preventing guest contributions here will help. My understanding: people still need to fill in their email when they register as guests, so admins already know "who is coming to the event".

**Comment by SudharakaP** [comment]
> @SudharakaP I don't understand how preventing guest contributions here will help. My understanding: people still need to fill in their email when they register as guests, so admins already know "who is coming to the event".

@Betree: Ah yes, that makes sense; I thought that the organizer might not have the same information for guest contributions. So yes, let us not have this checkbox to prevent guest contributions. 

Maybe it would be nice to have a discussion on this with @BenJam to make sure we are all on the same page on this so we can move forward with this. 😃

**Comment by robindpw** [comment]
Hey Robin a user of the platform here,

I requested a similar thing by email and got redirected here.  

For me the proposal to block people buying multiple tickets would be a life saver. This would increase the transparency in my opinion because you are then enforcing people to log in ( if my basic knowledge is right).  For my collective specifically, this makes sense because i dont really care about anonymous guests as they caused me problems before.

I also gave the comment into mail to add a text field so users could fill in stuff but like told before this isn't a good idea for all kind of reasons.

Thx for the beautiful tool already and your quick support ❤️

**Comment by BenJam** [comment]
tbh I am fine either way but blocking users from multiple registrations seems like a necessary pre-condition to allowing multiple registrations for others that may/may not be on the platform as part of a group ticket purchase.

**Comment by SudharakaP** [comment]
@Betree : I think as a MVP to solve this we could implement what I highlighted in https://github.com/opencollective/opencollective/issues/5086#issuecomment-1264029870 above (point 1),

>  Store if it's single or multiple tickets within the Tier.data column ({ singleTicket: true }) and then make sure we don't show the quantity selector when booking if it's a single ticket.

And I think this is all we need to do at this point to keep this simple.

**Comment by SudharakaP** [comment]
> @SudharakaP How are collective admins going to edit this `singleTicket` flag?

Sorry, I think I should have explained myself clearly. We add another option to the ticket type dropdown,

![image](https://user-images.githubusercontent.com/12435965/196218638-13fd9563-5797-4c0c-bebb-e3f7c57158cc.png)

We can name it, `Single Ticket per Order`. And toggling this will set the `singleTicket` flag.

**Comment by Betree** [comment]
@SudharakaP That works for me! Just be aware that #3438 is moving on quickly; the old edit tiers page is going to be replaced in a near future. The fix should ideally cover both.

**Comment by Betree** [comment]
@SudharakaP Please don't use this approach of changing the tier types. I was afraid that this would come up while reading the specs above, but I strongly think we should stay on the `data.singleTicket` flag approach. Why:
- Regardless of whether you allow one or more, you're still ordering tickets. We should not complexity our SQL queries with 2 conditions while we don't really care about it in most cases.
- There are countless places depending on the tier type `TICKET` - frontend, API, taxes, metabase queries. I'd rather have a "messy hack" in the frontend than messy hacks in all these projects. The simplicity of this issue is just not worth introducing such a big refactor.
- We're trying to solve a frontend concern by changing the data model, but honestly, I don't even think having this as a tier type in the UI is a good thing; to me, it should ultimately be a checkbox.

**Comment by SudharakaP** [comment]
> @SudharakaP Please don't use this approach of changing the tier types. I was afraid that this would come up while reading the specs above, but I strongly think we should stay on the `data.singleTicket` flag approach. Why:
> 
>     * Regardless of whether you allow one or more, you're still ordering tickets. We should not complexify our SQL queries with 2 conditions while we don't really care about it in most cases.
> 
>     * There are countless places depending on the tier type `TICKET` - frontend, API, taxes, metabase queries. I'd rather have a "messy hack" in the frontend than messy hacks in all these projects. The simplicity of this issue is just not worth introducing such a big refactor.
> 
>     * We're trying to solve a frontend concern by changing the data model, but honestly, I don't even think having this as a tier type in the UI is a good thing; to me, it should ultimately be a checkbox.

@Betree : That makes sense. I've opened two new PRs using the `data.singleTicket` method. And also I used the `checkbox` as your suggested instead of trying to put everything in the `type` drop-down. Let me know if there's any improvements to be made. I've also added this to the new Tier section as well, although I think there might be some bug existing in the new tiers page. For example it does show Tickets by default but when you click on a ticket the type is blank. 

![image](https://user-images.githubusercontent.com/12435965/197074638-f8ce33b6-cfe1-41cb-8b65-f2cb9d812295.png)

Are we showing both tickets and other (non-ticket) tiers in the same page within the new tiers? 🤔

**Comment by Betree** [comment]
> Are we showing both tickets and other (non-ticket) tiers in the same page within the new tiers? thinking

Currently, yes. But that's something we want to change for the public release: https://opencollective.slack.com/archives/C044GNQ7WBX/p1664539432219079

**Comment by Betree** [comment]
@BenJam I'm not convinced that this should be the default. Taking EventBrite as a reference (but I'm pretty sure it's the case of most other event platforms) they let you pick a number of tickets when you place your order.
